### PR TITLE
chore(xplat): Add support for nested selectors to Griffel to StyleX converter

### PR DESCRIPTION
**Pull request by behowell**
ℹ️ Note this is being merged into the `xplat` branch

## Previous Behavior

The convertGriffelToStyleX platform adapter didn't support nested selectors.

## New Behavior

The converter now adapts the griffel format of nested selectors:
```ts
{
  color: 'red',
  ':hover': {
    color: 'green',
    '@media (forced-colors: active)': {
       color: 'highlighttext',
     },
  },
}
```
into the stylex format of nested selectors:
```ts
{
  color: {
    default: 'red',
    ':hover': {
      default: 'green',
      '@media (forced-colors: active)': 'highlighttext',
    },
  },
}
```

### CODE REVIEW

File: packages/react-components/react-platform-adapter/src/styling/convertGriffelToStyleX.native.ts [hunk 1]
```diff
@@ -1,85 +1,131 @@
...
   -readonly [P in keyof StyleXStyle]: StyleXStyle[P];
 };
 
-type ComplexValueType = {
-  default?: string | number;
-  [pseudoClass: `:${string}`]: string | number;
-  [atRule: `@${string}`]: string | number;
+type ValueWithSelectors = {
+  default?: string | number | ValueWithSelectors | null;
+  [pseudoClass: `:${string}`]: string | number | ValueWithSelectors;
```
Comment (JasonVMo) [inline packages/react-components/react-platform-adapter/src/styling/convertGriffelToStyleX.native.ts:11]: Interesting, didn't realize index signatures could be added this way.

File: packages/react-components/react-platform-adapter/src/styling/convertGriffelToStyleX.native.ts [hunk 2]
```diff
@@ -1,85 +1,131 @@
...
-      }
+    if (typeof value === 'string' || typeof value === 'number') {
+      strictDomStyle[property] = makeValueWithSelectors(selectors, value, strictDomStyle[property]);
+    } else if (Array.isArray(value)) {
+      console.warn('Unsupported array value for ', key);
+    } else if (isSelector(key) && isValueWithSelectors(value)) {
+      mergeSelectorValues([...selectors, key], value, strictDomStyle);
     } else {
-      throw new Error(`Unsupported nested selector: '${selector}' '${key}'`);
+      console.warn(`Unsupported selector `, [...selectors, key]);
```
Comment (JasonVMo) [inline packages/react-components/react-platform-adapter/src/styling/convertGriffelToStyleX.native.ts:93]: If I follow the logic now, we will emit warnings with unsupported selectors but not end up passing them to stylex. I think that is better than having the hard errors.

**Review by JasonVMo — APPROVED** [review]
Good catch on this one, I hadn't noticed the signature differences between stylex and griffel for these.

File: packages/react-components/react-platform-adapter/src/styling/convertGriffelToStyleX.native.ts (hunk 2 above)
Comment (behowell) [reply packages/react-components/react-platform-adapter/src/styling/convertGriffelToStyleX.native.ts:93]: Yeah, we'll need a longer-term solution to catch and fix these cases at development time, like a lint rule or something. But for now, this at least makes it so it doesn't crash with an invalid selector.

**Pull request merged**